### DISCUSSION TITLE
PATROL: Flutter Multi Modularisation Support?

### DISCUSSION THREAD

**Discussion by RashmiShashindra** [leancodepl/patrol#1628] [Q&A] [answered]
I am just starting to use Patrol for the Integration Tests. 

I need to integrate the patrol tests in multi modularity. Does it support multi modularity? If so can you please share me a sample code or a git repo to that please?

**Comment by bartekpacia** [comment]
Hi @RashmiShashindra, please clarify what you mean.

**Reply by RashmiShashindra** [reply]
Hi @bartekpacia , Thanks for getting back.

So our development code has the each package for each feature. (Multi Modularity) So can I write the integration_test s inside each feature package? Does that support in Patrol? Or should I still create the integration_test in root and write the tests?

**Reply by bartekpacia** [reply, marked as answer]
(By Patrol I'm assuming you mean [Patrol's native automation feature](https://patrol.leancode.co/native/overview)).

Patrol is for E2E testing your real app on a real device. So I think you should create the `integration_test` directory in your app's main package (where the Dart entrypoint is) and write tests there.

**Reply by RashmiShashindra** [reply]
@bartekpacia Yes sure. I implemented this in main package and it works for me. Thanks!
